docs/usage/configuration-options.md

```diff
@@ -1412,14 +1412,21 @@ This feature can be used to refresh lock files and keep them up-to-date.
 "Maintaining" a lock file means recreating it so that every dependency version within it is updated to the latest.
 Supported lock files are:
```

Comment:
Updating the list manually has some drawbacks:

- I can't know all possible lockfiles and their matching managers
- For me, finding the lockfiles in the package manager official docs can be hard
- The programmers that write the manager code know the supported lockfiles
- putting new entries in the list is likely to mess up the alphabetical ordering by accident

Can we automate the list?

How about this:

1. Each package manager has a new constant which has an array with strings of valid lock files for that manager, and the name of the package manager
2. Docs generating code grabs alls the consts, and creates one A-Z sorted lists of supported lockfiles formats plus the package manager name

So the final format:

```markdown
List of supported lockfiles:

- `lockfile format` (manager name)
- `lockfile format 1`, `lockfile format 2` (for manager that supports multiple lockfiles)
- and so on...
```

Reply:
sounds good, please open an issue to track

Reply:
Here's the tracking issue:

- #19340